pkgs/development/compilers/riscv-gnu-toolchain/default.nix

```diff
@@ -0,0 +1,100 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchurl
+
+, gcc12
+, glibc
+, newlib
+, gdb
+, qemu
+, musl
+, spike
+, riscv-pk
+
+, autoconf
+, curl
+, automake
+, python3
+, libmpc
+, gawk
+, bison
+, flex
+, texinfo
+, gperf
+, libtool
+, expat
+, gmp
+, mpfr
+, flock
+, zlib
+
+, srcs ? {
+  binutils = {
+    url = "mirror://gnu/binutils/binutils-2.39.tar.bz2";
+    hash = "sha256-2iSoT+8iAQLdJAQt8G/eqFHCYUpTd/hu/6KPM7exYUg=";
+  };
+  dejagnu = {
+    url = "mirror://gnu/dejagnu/dejagnu-1.6.3.tar.gz";
+    hash = "sha256-h9rvrNeVi0pp+IxoVtvRY0JhljxBQHnQw3H1ic1mouM=";
+  };
+  gcc = gcc12.cc.src;
+  glibc = glibc.src;
+  newlib = newlib.src;
+  gdb = gdb.src;
+  qemu = qemu.src;
+  musl = musl.src;
+  spike = spike.src;
+  pk = riscv-pk.src;
+}
+
+, isa ? "rv64gc"
+, withLinux ? false
+}:
+let
+
+extract = file: stdenv.mkDerivation {
+  name = file.name + "-unpacked";
+  src = file;
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
+  installPhase = "cp -r . $out";
+  dontFixup = true;
+};
+
+srcs_fetched = lib.mapAttrs (name: value: if value ? outPath then value else fetchurl value) srcs;
+srcs_extracted = lib.mapAttrs (name: value: extract value) srcs_fetched;
```

Comment:
For which src are we doing this? Maybe we can just change that one package to avoid this.

Reply:
hree of them use fetchFromGitHub, and I'm not aware of a way to make that use fetchurl instead of fetchzip

Reply:
Why do we want to use fetchurl for github sources or am I mixing things up?

Reply:
some use `fetchFromGitHub`, and some use `fetchurl`. This means we have a mix of compressed and uncompressed sources